lib/syntax_tree/node.rb

```diff
@@ -2150,30 +2167,44 @@ def call(q)
     SEPARATOR = Separator.new.freeze
 
     def format(q)
-      q.text("|")
-      q.group do
-        q.remove_breaks(q.format(params))
-
-        if locals.any?
-          q.text("; ")
-          q.seplist(locals, SEPARATOR) { |local| q.format(local) }
+      if pipe?
+        q.text("|")
```

Comment:
Can you move the formatting of the pipes up to the parent class? That should be in `BlockNode`. That way the formatting here can be consistent and you shouldn't have to track `pipe?` at all.